TreeDB/collections/column_hnsw_search_pack_writer.go

```
@@ -302,21 +316,501 @@ func validateColumnHNSWSearchPackStateAssetIfPresentWithMode(rootDir string, cfg
+			return err
+		}
+	}
+	directoryLength := hnswPackU64(prefix, columnHNSWSearchPackHeaderDirectoryLengthOffset)
+	if directoryLength > uint64(ref.Length-int64(headerSize)) || directoryLength > uint64(math.MaxInt-headerSize) {
+		return fmt.Errorf("collections: hnsw search pack directory length=%d outside asset", directoryLength)
+	}
+	metadata := make([]byte, headerSize+int(directoryLength))
+	if err := readColumnHNSWSearchPackFileAt(file, ref.Offset, metadata); err != nil {
+		return err
+	}
+	pack, opts, err := decodeColumnHNSWSearchPackEnvelopeMetadataWithContext(nil, metadata, uint64(ref.Length), columnHNSWSearchPackDecodeOptions{ExpectedBaseIdentity: columnHNSWSearchPackBaseIdentity{
```

golangci-lint (2.12.2)

[error] SA1012: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use

(staticcheck)